### DISCUSSION TITLE
Use custom toolset LLVM-vs2014

### DISCUSSION THREAD

**Discussion by yurybura** [microsoft/vcpkg#17764] [Ideas]
Unable to build zlib (for example) with custom triplet (x86_64-windows-llvm-vs2014.cmake):
```
set(VCPKG_TARGET_ARCHITECTURE x64)
set(VCPKG_PLATFORM_TOOLSET LLVM-vs2014)
set(VCPKG_CRT_LINKAGE dynamic)
set(VCPKG_LIBRARY_LINKAGE static)
if(PORT STREQUAL "gtest")
    set(VCPKG_LIBRARY_LINKAGE dynamic)
endif()
if(PORT STREQUAL "boost")
    set(VCPKG_LIBRARY_LINKAGE dynamic)
endif()
```
Output:
d:\vcpkg>vcpkg version
Vcpkg package management program version 0.0.100-2017-12-06-29163e1566403c371a5af48c8dbf3d96da32b8d1

d:\vcpkg>vcpkg install zlib:x86_64-windows-llvm-vs2014
The following packages will be built and installed:
    zlib:x86_64-windows-llvm-vs2014
Starting package 1/1: zlib:x86_64-windows-llvm-vs2014
Building package zlib:x86_64-windows-llvm-vs2014...
Could not find Visual Studio instace with LLVM-vs2014 toolset.

**Comment by ras0219-msft** [comment]
Thank you for posting this issue!

This is unfortunately not currently supported, but we'd be happy to accept a PR that fixes the issue!

**Comment by Neumann-A** [comment]
This is nowadays supported via the workaround:
```
set(VCPKG_LOAD_VCVARS_ENV ON) # Setting VCPKG_CHAINLOAD_TOOLCHAIN_FILE deactivates automatic vcvars setup so reenable it!
if(DEFINED VCPKG_PLATFORM_TOOLSET) # Tricks vcpkg to load vcvars for a VCPKG_PLATFORM_TOOLSET which is not vc14[0-9]
    set(VCPKG_PLATFORM_TOOLSET ClangCL)
endif()
```

see https://github.com/Neumann-A/my-vcpkg-triplets/blob/master/x64-win-llvm.cmake
